stan/math/rev/mat/functor/integrate_1d_tsc.hpp

```diff
@@ -22,83 +21,86 @@ namespace stan {
 namespace math {
 
 /**
- * Return the numerical integral of a function f given its gradient
- * g.
+ * Return the numerical integral of the specified function and
+ * gradient function, within the specified integration limits, given
+ * parameters, data, and tolerances for integration.
  *
- * This function uses the algorithm of integrate_1d_tscg for
- * numerical integration.
+ * This function uses the algorithm of <code>integrate_1d_tscg</code>
+ * for numerical integration.
  *
  * @tparam T Type of f.
  * @tparam G Type of g.
+ * @tparam T_param Type of param.
+ * @tparam T_x_r Type of x_r.
+ * @tparam T_x_i Type of x_i.
+
  * @param f a functor with signature
- * double (double, std::vector<T_param>) or with signature
- * double (double, T_param) where the first argument is one being
- * integrated and the second one is either an extra scalar or vector
- * being passed to f.
+ * <code>double(double, C_param, data C_x_r, data C_x_i, ostream&)</code>
+ * where `C_param` is `T_param` or `vector<T_param>`, `C_x_r` is
+ * `T_x_r` or `vector<T_x_r>`, and `C_x_i` is `T_x_i` or
+ * `vector<T_x_i>`, with the first argument being integrated, the
+ * second being parameters, the next two being fixed data, and the
+ * last being a stream for messages.
  * @param g a functor with signature
- * double (double, std::vector<T_param>, int, std::ostream&) or with
- * signature double (double, T_param, int, std::ostream&) where the
- * first argument is being integrated and the second one is
+ * double(double, std::vector<T_param>, std::vector<T_x_r>,
+ * std::vector<T_x_i>, int, std::ostream&) or with signature double
+ * (double, T_param, T_x_r, T_x_i, int, std::ostream&)
+ * where the first argument is being integrated and the second one is
  * either an extra scalar or vector being passed to f and the
- * third one selects which component of the gradient vector
+ * fifth one selects which component of the gradient vector
  * is to be returned.
  * @param a lower limit of integration, must be a finite double.
  * @param b upper limit of integration, must be a finite double.
  * @param tre target relative error.
  * @param tae target absolute error.
  * @param param additional parameters to be passed to f.
+ * @param x_r additional data to be passed to f.
+ * @param x_i additional integer data to be passed to f.
  * @param msgs stream.
  * @return numeric integral of function f.
  */
-template <typename F, typename G, typename T_param>
+template <typename F, typename G, typename T_param, typename T_x_r,
```

Comment:
We don't need template parameters for `T_x_r` and `T_x_i`. These should be removed and where used, just include the primitive types: `double` and `int`.